utils/misc.mli

```diff
@@ -314,3 +314,26 @@ val normalise_eol : string -> string
 val delete_eol_spaces : string -> string
 (** [delete_eol_spaces s] returns a fresh copy of [s] with any end of line spaces
    removed. Intended to normalize the output of the toplevel for tests. *)
+
+
+
+
+(* Hooks machinery:
+  [add_hook name f] will register a function that will be called on the
+    argument of a later call to [apply_hooks]. Hooks are applied in the
+    lexicographical order of their names.
+*)
+
+exception HookExn of exn
```

Comment:
We are designing a new API for external use, so let's make it as smooth as possible.  One could simply register the wrapper constructor in Misc and record the printer somewhere else (e.g. in Location).  I'd also use this wrapper, which would normally terminate the compiler, but not necessarily the toplevel.